### PR TITLE
add groupBy on collections to IR

### PR DESCRIPTION
**Pull request by catoverdrive**

### CODE REVIEW

File: src/main/scala/is/hail/asm4s/FunctionBuilder.scala [hunk 1]
```diff
@@ -296,7 +296,7 @@ class FunctionBuilder[F >: Null](val parameterTypeInfo: Array[MaybeGenericTypeIn
 
       cn.accept(cw)
       bytes = cw.toByteArray
-      // CheckClassAdapter.verify(new ClassReader(bytes), false, new PrintWriter(sw1))
+       CheckClassAdapter.verify(new ClassReader(bytes), false, new PrintWriter(sw1))
```
Comment (rcownie) [inline src/main/scala/is/hail/asm4s/FunctionBuilder.scala]: Is it intentional to have this enabled in the committed code ?  It looks to me
as though it's for debugging

File: src/main/scala/is/hail/expr/ir/Emit.scala [hunk 2]
```diff
@@ -467,9 +544,9 @@ private class Emit(
         val tarray = coerce[TArray](a.typ)
         val tti = typeToTypeInfo(typ)
         val eti = typeToTypeInfo(tarray.elementType)
-        val xmv = mb.newField[Boolean](name2+"_missing")
+        val xmv = mb.newField[Boolean](name2 + "_missing")
```
Comment (rcownie) [inline src/main/scala/is/hail/expr/ir/Emit.scala]: I have a weak preference for not mixing formatting changes into a substantive commit.

File: src/main/scala/is/hail/asm4s/FunctionBuilder.scala (hunk 1 above)
Comment (catoverdrive) [reply src/main/scala/is/hail/asm4s/FunctionBuilder.scala]: no, I accidentally left it uncommented from when I was debugging :) I'll take it back out.

Comment (catoverdrive) [reply src/main/scala/is/hail/asm4s/FunctionBuilder.scala]: (done)

**Pull request merged**